API/functions/index.js

```diff
@@ -63,15 +76,36 @@ messaging_api.get('/test', async (request, response) => {
 messaging_api.get('/getMessages', async (request, response) => {
   const req_key = request.get('auth');
   if (req_key == KEY) {
-    await response.json(await getMessages());
+    await response.json(await getMessages(request.query.uid));
   } else {
     response.status(401).send('Unauthorized');
   }
 });
 
-async function getMessages() {
-  const snapshot = await firestore.collection('test_users').doc(getUid()).get(); 
-  return snapshot.data().map;
+async function getMessages(uid) {
+  const snapshot = await firestore.collection('test_users').doc(uid).get(); 
+  return snapshot.data();
 }
 
+//delete messages
+messaging_api.get('/deleteMessages', async (request, response) => {
+  const req_key = request.get('auth');
+  if (req_key == KEY) {
+    await response.json(await deleteMessages(request.query.uid));
+  } else {
+    response.status(401).send('Unauthorized');
+  }
+});
+
+async function deleteMessages(uid) {
+  const snapshot = await firestore.collection('test_users').doc(uid).update({
+    "msgs": admin.firestore.FieldValue.delete()})
```

Comment:
Let's use the `USERS` constant instead of `firestore.collection('test_users')`. It will be easier when we need to change the location of the users collection from `test_users` to `users` or something else.

Reply:
Thanks for catching this. Fixed it.